Python/compile.c

```diff
@@ -3893,8 +3893,8 @@ compiler_dict(struct compiler *c, expr_ty e)
             ADDOP_I(c, DICT_UPDATE, 1);
         }
         else {
-            if (elements == 0xFFFF) {
-                if (!compiler_subdict(c, e, i - elements, i)) {
+            if (elements == USHRT_MAX) {
+                if (!compiler_subdict(c, e, i - elements, i + 1)) {
```

Comment:
I just made the minimum edit to fix the problem but I would just remove this checkpoint and simply keep incrementing `elements` and leave the last check fill the dict.